packages/jest-cli/src/runTest.js

```diff
@@ -13,74 +13,91 @@ import type {Path, Config} from 'types/Config';
 import type {TestResult} from 'types/TestResult';
 import type {Resolver} from 'types/Resolve';
 
+const fs = require('fs');
+const {getTestEnvironment} = require('jest-config');
 const BufferedConsole = require('./lib/BufferedConsole');
+const promisify = require('./lib/promisify');
 const {
   Console,
   NullConsole,
   setGlobal,
 } = require('jest-util');
 const getConsoleOutput = require('./reporters/getConsoleOutput');
 
+const testEnvironmentPragmaRegex = /@testEnvironment\s+(.*)/;
```

Comment:
Also, I was thinking we could call this `@jest-environment` instead. That way we can add more of these in the future with the same prefix :)